### ISSUE TITLE
Server Crashinng on Startup after loading save/creating world

### ISSUE THREAD

**Issue by aaronproctor** [wolveix/satisfactory-server#136]
**Describe the Bug**
Server is crashing and docker container is stopping while trying to load world

**Your Runtime Command or Docker Compose File**
Not sure what you need. I'm using docker desktop

**System Specs (please complete the following information):**
===== START ISSUE REPORT =====
OS:  Linux server 5.17.5-76051705-generic #202204271406~1655476786~22.04~62dd706 SMP PREEMPT Fri Jun 17 16 x86_64 x86_64 x86_64 GNU/Linux
CPU: AMD A10-7800 Radeon R7, 12 Compute Cores 4C+8G
RAM: 8GB/14GB
HDD: 22GB/220GB (11% used)
===== END ISSUE REPORT =====

**Comment by aaronproctor** [comment]
Using the server manager I download the save and load it that way. I also tried creating a new world which have the same result

**Comment by wolveix** [comment]
You say you download the world, did you mean upload? The intended workflow is to start the container, access the server via the in-game server manager, load the save, then join once it's loading.

If you are doing that, you're finding that the container is crashing and restarting?

**Comment by aaronproctor** [comment]
Sorry for the late reply. Yes I mean upload. Following those instructions the server tries to start then crashes and does not restart. I've since just used the steam version of the server but I'd be glad to help figure out if this problem is an issue on my end or yours

**Comment by wolveix** [comment]
All good! Would you mind sharing your save with me? I'm curious as to whether it's something weird with your save. Alternatively, try creating a fresh save in your client and then uploading it

**Comment by aaronproctor** [comment]
Sure! I did try making a new save too but would run into the same issue
[GOLDxGUNS_autosave_1.zip](https://github.com/wolveix/satisfactory-server/files/9176478/GOLDxGUNS_autosave_1.zip)

**Comment by Raik0** [comment]
I would like to add that I too am getting this issue. I am getting it with both uploaded save and new session creation.
I've recently spun up a new Truenas Scale box.

**Comment by wolveix** [comment]
Hey guys, sorry for the lack of communication. I've been too busy to have a chance to properly delve deeper into this.

@preimmortal that's interesting to hear. The old version of the Docket image (pre-CSS' official server implementation) did indeed require a substantial amount of RAM; however, the newer build didn't from my observations. When did you first encounter the high memory requirement?

**Comment by DerZugvogel** [comment]
Have the same Issue. Same Error message. Rather creating or loading a save uploaded via server-manager nor loading a save uploaded via ftp. Nothing worked. Server has Intel Xeon, Debian 10 and 16gb RAM.

**Comment by wolveix** [comment]
@DerZugvogel could you submit the information requested in the issue report template, please? Runtime command/compose file, log file, and system specs (using the `DEBUG` env variable).

**Comment by DerZugvogel** [comment]
Yeah, sure. 

**My start command without debug env var**
`docker run -d --name=satisfactory-server -h satisfactory-server -m 14336m -e MAXPLAYERS=6 -e PGID=1000 -e PUID=1000 -e STEAMBETA=false -e MULTIHOME=202.61.240.21 -v /home/*user*/satisfactory-data:/config -p 7777:7777/udp -p 15000:15000/udp -p 15777:15777/udp wolveix/satisfactory-server:latest`

Tested without multihome too. Tried to set more memory with the -m variable, because [FactoryGame.log](https://github.com/wolveix/satisfactory-server/files/9793331/FactoryGame.log) says
`[2022.10.16-00.00.54:676][449]LogCore: Error: appError called: Fatal error: [File:D:/ws/SB-220602145101-842/UE4/Engine/Source/Runtime/Core/Private/Unix/UnixPlatformMemory.cpp] [Line: 1031] 
Ran out of memory allocating 420024320 bytes with alignment 4096
<< callstack too long >>`
It somehow uses only 4096mb memory

[Diagnostics.txt](https://github.com/wolveix/satisfactory-server/files/9793332/Diagnostics.txt)

Another Problem: With the debug env-var the server doenst even start. It don't show up on docker container list.

I'm really new to the docker world and not really a professional in server topics. Sorry for inconvinience

**Comment by wolveix** [comment]
Yeah, `MULTIHOME` does nothing for this image. You're after `SERVERIP`, though you almost definitely do not need it.

Seems like a RAM issue, as the latest release does use a significant amount of RAM (see #143), generally between 10GB - 16GB (depending on the world). Does the container fully crash? Or do you just get disconnected?

**Comment by DerZugvogel** [comment]
Ahh, okay thx. Good to know.

Yes, my World uses on my other Server about 11GB. 16 are installed and the CrashClient Log says it detects 16gb and a AMD EPYC. My old server provides bad tps, so I want to switch to my vserver.
The container fully crashes. I tried to create a new world and to load my normal save
